Brep application

tteschemacher opened the pull request:
@KratosMultiphysics/nurbs-breps - updated code for closest point search. 
Definition of interface for other application needed.

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into Brep_Application

tteschemacher committed:
First Commit NurbsApplication

Corrections in IGAStructuralMechanicsApplication and first commit of
NurbsBrepApplication.

Commit:
Merge branch 'Brep_Application' of https://github.com/KratosMultiphysics/Kratos into Brep_Application

tteschemacher committed:
Naming Changes

Changes in the naming of the Application

tteschemacher committed:
Merge branch 'Brep_Application' of https://github.com/KratosMultiphysics/Kratos into Brep_Application

Commit:
bug fix in NurbsShape

tteschemacher committed:
Geometry Read In

Set up of structure. Now geometry as JSON can be read in - already
depending the right model part.

Commit:
Tabs replaced by white spaces

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into Brep_Application

Commit:
Tabs replaced by white spaces

Commit:
Formating stuff

Commit:
automatic search for new files in NurbsBrepApplication

Commit:
replaced some additional tabs by white spaces

Commit:
removed unnecessary dependency

tteschemacher committed:
Updated to new Geometry JSON

New JSON can now be read.

RiccardoRossi committed:
code review of the preliminary design

tteschemacher committed:
First Code Review Corrections

Commit:
some changes in CMakeLists NurbsBrepApplication

tteschemacher committed:
Kratos Paramaters class used now

tteschemacher committed:
Merge branch 'Brep_Application' of https://github.com/KratosMultiphysics/Kratos into Brep_Application

tteschemacher committed:
New geometry file

Commit:
remove unnecessary file

Commit:
Merge branch 'Brep_Application' of https://github.com/KratosMultiphysics/Kratos into Brep_Application

tteschemacher committed:
Mapping

Commit:
Merge branch 'Brep_Application' of https://github.com/KratosMultiphysics/Kratos into Brep_Application

tteschemacher committed:
Closest Point Projection

tteschemacher committed:
Python file

tteschemacher committed:
Visual Code Launcher

tteschemacher committed:
Added Missing CPPs

Commit:
Merge branch 'Brep_Application' of https://github.com/KratosMultiphysics/Kratos into Brep_Application

tteschemacher committed:
Simple Geometry File

tteschemacher committed:
Add Shape Functions to Nodes

pooyan-dadvand reviewed (changes requested):
I have noticed that you have an Edge and Face classes which would be in conflict with ones I'm using for mesh (in my branch). 

Would you please change their name to indicate that they are for NURBS?

RiccardoRossi commented on applications/IGAStructuralMechanicsApplication/custom_conditions/meshless_lagrange_coupling_condition.cpp:
```diff
@@ -90,26 +89,37 @@ void MeshlessLagrangeCouplingCondition::CalculateLocalSystem(MatrixType& rLeftHa
+			// |H 0  |
 			//lambda in X
-			rLeftHandSideMatrix(i * 3, 3 * j)         = NN + Phi_r_Lambda((i - number_of_points) * 3 )  * Phi_r(j * 3);//Phi_r_Lambda((i - number_of_points)*3)*Phi_r(j*3); ShapeFunctionsN[i] * Phi_r(j * 3);// 
+			rLeftHandSideMatrix(i * 3, 3 * j) = NN +Phi_r_Lambda((i - number_of_points) * 3)  * Phi_r(j * 3);//Phi_r_Lambda((i - number_of_points)*3)*Phi_r(j*3); ShapeFunctionsN[i] * Phi_r(j * 3);// 
```
i would suggest to define 
const ibase = i*3;
const jbase = j*3

same for 
(i - number_of_points) * 3

RiccardoRossi commented on applications/IGAStructuralMechanicsApplication/custom_conditions/meshless_lagrange_coupling_condition.cpp:
```diff
@@ -132,20 +142,18 @@ void MeshlessLagrangeCouplingCondition::CalculateLocalSystem(MatrixType& rLeftHa
 
-	double JGeometrictoParameter;
-	MappingGeometricToParameterMasterElement(DN_DeMaster, aux, JGeometrictoParameter);
+	array_1d<double, 2> localTrimTangentsMaster;
```
in one single line
array_1d<double, 2> localTrimTangentsMaster = localTrimTangents;

RiccardoRossi commented on applications/IGAStructuralMechanicsApplication/custom_utilities/IntegrationDomainModeler.h:
```diff
@@ -0,0 +1,227 @@
+/**
+ * Typedefs for search
+ */
+
```
do not make such typdefs in the global kratos scope

RiccardoRossi commented on applications/IGAStructuralMechanicsApplication/custom_utilities/IntegrationDomainModeler.h:
```diff
@@ -0,0 +1,227 @@
+		///@name Life Cycle 
+		///@{ 
+
+		void SetUp(
```
why don't you add this in the constructor?

RiccardoRossi commented on applications/IGAStructuralMechanicsApplication/examples/OilsumpLagrange/ImportModelPart.py:
```diff
@@ -0,0 +1,428 @@
+		dimension = 2
+
+		with open(self.JsonFileName) as JsonFile:
+			nodes_list = json.load(JsonFile)['nodes']
```
!!!! here you are parsing a lot of times the json file.
 
    with open(self.JsonFileName) as JsonFile:
       parsed_json = json.load(JsonFile)
    
    nodes_list = parsed_json['nodes']
    dim2_elements_list = parsed_json['2d_elements']

RiccardoRossi commented on applications/IGAStructuralMechanicsApplication/examples/OilsumpLagrange/ImportModelPart.py:
```diff
@@ -0,0 +1,428 @@
+
+
+		#--- CONDITIONS ---#
+		with open("ProjectParameters.json") as JsonFile:
```
again...do not reread it!!

RiccardoRossi commented on applications/IGAStructuralMechanicsApplication/examples/OilsumpLagrange/ImportModelPart.py:
```diff
@@ -0,0 +1,428 @@
+			
+			Condition = 0
+		
+			with open("ProjectParameters.json") as JsonFile:
```
same. read nly once

RiccardoRossi commented on applications/NurbsBrepApplication/custom_utilities/BrepModelGeometryReader.cpp:
```diff
@@ -0,0 +1,286 @@
+      EdgesVector edges_vector;
+
+      // loop over faces
+      for (int i = 0; i < brep_json["faces"].size(); i++)
```
use iterators. it will be WAY faster... with this i mean orders of magnitude faster depending on your access pattern

tteschemacher committed:
Renaming of all brep classes

Face -> BrepFace
Edge-> BrepEdge
...

pooyan-dadvand reviewed (dismissed):
👍

tteschemacher committed:
Update IGA Structural Mechanics Problem

RiccardoRossi commented on applications/IGAStructuralMechanicsApplication/custom_conditions/meshless_base_coupling_condition.cpp:
```diff
@@ -208,10 +208,8 @@ void MeshlessBaseCouplingCondition::CaculateRotation(const Matrix &ShapeFunction
-	array_1d<double, 3> g10;
-	array_1d<double, 3> g20;
-	array_1d<double, 3> g30;
+	int number_of_points = ShapeFunctionDerivatives.size1();
```
make this const unsigned int

RiccardoRossi commented on applications/IGAStructuralMechanicsApplication/custom_conditions/meshless_load_condition.cpp:
```diff
@@ -112,6 +112,7 @@ namespace Kratos
 			direction = g3 / dArea;
+			//KRATOS_WATCH(direction)
 		}
 		else if (SURFACE_DEAD == 1)
```
why using capitals here? (we normally use capitals only for kratos variables)

RiccardoRossi commented on applications/IGAStructuralMechanicsApplication/custom_elements/meshless_shell_element.cpp:
```diff
@@ -127,119 +126,87 @@ void MeshlessShellElement::Initialize()
+
 	//calculating the total area
-	mTotalDomainInitialSize = mDetJ0 * integration_weight;
+	mTotalDomainInitialSize = mDetJ0 * integration_weight; //RAUS SCHMEI?EN
```
please do not use german (and less so, "strange" characters) in the comments

RiccardoRossi commented on applications/IGAStructuralMechanicsApplication/custom_elements/meshless_shell_element.cpp:
```diff
@@ -583,45 +340,13 @@ void MeshlessShellElement::CalculateAndAddNonlinearKm(
 				for (unsigned int j = 0; j < check; j++)
 				{
 					K(3*n + i, 3*m + j) += (SD[0] * B11(3*n + i, 3*m + j) + SD[1] * B22(3 * n + i, 3 * m + j) + SD[2] * B12(3 * n + i, 3 * m + j))*weight;
```
it may help readability to haave something like

unsigned int base_n = 3*n
unsigned int base_m = 3*m

RiccardoRossi commented on applications/IGAStructuralMechanicsApplication/custom_elements/meshless_shell_element.cpp:
```diff
@@ -1108,34 +820,37 @@ void MeshlessShellElement::CalculateAll(
-	Vector StrainDeformation;
-	Vector CurvatureDeformation;
+	//Local Cartesian Foreces and Moments
+	Vector ForceVector_in_Q_coordinates = ZeroVector(3);
```
you don't need to first initialize this to zero. i mean it would be shorter and run faster to do 

Vector ForceVector_in_Q_coordinates = prod(trans(DMembrane), StrainVector_in_Q_coordinates);

RiccardoRossi reviewed (dismissed):
it looks in good shape to me...i made some comments which you may want to look into, however no blocker

tteschemacher committed:
Multipatch Coupling

Possible Operations: Coupling and application of weak boundary
conditions.

Commit:
Merge branch 'Brep_Application' of https://github.com/KratosMultiphysics/Kratos into Brep_Application

Commit:
faster debugging with python in release mode

Commit:
Merge branch 'master' of https://github.com/KratosMultiphysics/Kratos into Brep_Application

Conflicts:
	applications/CMakeLists.txt

d-herreroadan committed:
Minor changes to line 240

variables reference definition

pooyan-dadvand commented:
Just a very minor comment,

In general I would avoid adding the author for each function or method. One can find out it by using the blame mechanism of the GIT. It increases the number of the lines in a file and make looking to the real code more difficult without being necessary. Also they make confusion and conflicts when different developers touch the same part. There are many projects that completely avoid it. Nevertheless the Kratos permits it in the header info.

dbaumgaertner commented:
Oh, I was not aware, that my name sneaked in here. I agree that adding authors to every function just causes clutter and does not make sense at all. 

Whenever it appears in a meaningful context, however, I will make sure to adjust it to only an Ascii writing.

RiccardoRossi commented:
@dbaumgaertner , thank you :-)

thing is that the umlaud (not sure of how to spell it) looks like a strange character depending on the locale

cheers

d-herreroadan committed:
Minor changes + NurbsBrepApplication (lower case)

NURBSBrepApplication swapped for NurbsBrepApplication , in "imports" and in CMakeLists

tteschemacher committed:
Trimming

tteschemacher committed:
Upadate of IGA Structural Mechanics

tteschemacher committed:
Merge branch 'Brep_Application' of https://github.com/KratosMultiphysics/Kratos into Brep_Application

# Conflicts:
#	applications/NurbsBrepApplication/CMakeLists.txt
#	applications/NurbsBrepApplication/custom_utilities/brep/BrepFace.cpp
#	applications/NurbsBrepApplication/custom_utilities/brep/BrepFace.h
#	applications/NurbsBrepApplication/custom_utilities/knot_span/KnotSpan2d.h

tteschemacher committed:
New example

Example Bended Plate has to trimmed patches which are coupled.

RiccardoRossi commented:
@tteschemacher are u going to merge this? because the more time passes the more conflicts will appear to be resolved...

tteschemacher committed:
Projection improved

-Bugfix in projection
-Tripod exmaple attached

RiccardoRossi commented:
@tteschemacher are u pushing this to the repo?

RiccardoRossi commented:
@tteschemacher, is this branch active?

tteschemacher commented:
@RiccardoRossi This branch is still in use.
As soon as the test cases are active we can push it to the repo.

tteschemacher committed:
Bug fix

Compilation errors in linux fixed

tteschemacher committed:
Case Sensitives

Changes in case sensitive spellings.

loumalouomega commented:
@tteschemacher is this ready to merge?

RiccardoRossi commented:
@KratosMultiphysics/nurbs-breps 

is this issue stil relevant? if not i would close it

roigcarlo commented:
@tteschemacher Is this still alive?

tteschemacher committed:
updates

tteschemacher committed:
updates

tteschemacher committed:
Merge branch 'master' into Brep_Application

# Conflicts:
#	applications/CMakeLists.txt
#	applications/IGAStructuralMechanicsApplication/custom_elements/meshless_shell_element.cpp
#	applications/IGAStructuralMechanicsApplication/custom_elements/meshless_shell_element.h
#	applications/MeshingApplication/custom_processes/internal_variables_interpolation_process.h
#	applications/MeshingApplication/custom_processes/metrics_hessian_process.h
#	applications/applications_interface.py
#	embedded_python/krun_main.cpp
#	external_libraries/amgcl/util.hpp
#	kratos/processes/compute_nodal_gradient_process.h

tteschemacher committed:
updates with bug

philbucher committed:
Revert changes in Meshingapp

philbucher committed:
Revert changes in core

philbucher committed:
IGA and NURBS App compile

philbucher committed:
revert more changes
